src/borg/archive.py

```diff
@@ -1692,7 +1692,20 @@ def __init__(self):
         self.error_found = False
         self.possibly_superseded = set()
 
-    def check(self, repository, repair=False, first=0, last=0, sort_by="", match=None, verify_data=False):
+    def check(
+        self,
+        repository,
+        repair=False,
+        first=0,
+        last=0,
+        sort_by="",
+        match=None,
+        older=None,
+        oldest=None,
+        newer=None,
+        newest=None,
```

Comment:
guess you need to extend the boolean value computation of `self.check_all` to also consider these new parameters which can also make it not checking all archives, but only some.

Reply:
(done)